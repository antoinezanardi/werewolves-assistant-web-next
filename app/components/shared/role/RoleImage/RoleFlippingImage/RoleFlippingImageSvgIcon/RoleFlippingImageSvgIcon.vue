<template>
  <NuxtImg
    id="role-flipping-image-svg-icon"
    :alt="$t('components.RoleFlippingImageSvgIcon.iconAlt')"
    class="absolute animate__bounceIn bottom-4 right-2"
    :height="svgSize"
    placeholder="/svg/misc/ripples.svg"
    :placeholder-class="`w-[${svgSize}] h-[${svgSize}]`"
    :src="svgIconPath"
    :width="svgSize"
  />
</template>

<script setup lang="ts">
import type { RoleFlippingImageSvgIconProps } from "~/components/shared/role/RoleImage/RoleFlippingImage/RoleFlippingImageSvgIcon/role-flipping-image-svg-icon.types";

defineProps<RoleFlippingImageSvgIconProps>();

const svgSize = "75";
</script>