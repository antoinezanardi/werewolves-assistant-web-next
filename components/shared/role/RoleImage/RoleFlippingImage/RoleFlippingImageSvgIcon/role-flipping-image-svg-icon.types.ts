type RoleFlippingImageSvgIconProps = {
  svgIconPath: string;
};

export type { RoleFlippingImageSvgIconProps };