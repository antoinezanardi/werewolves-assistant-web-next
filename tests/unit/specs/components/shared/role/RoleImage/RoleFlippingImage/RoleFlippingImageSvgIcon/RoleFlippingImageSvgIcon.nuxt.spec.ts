import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import type { RoleFlippingImageSvgIconProps } from "~/components/shared/role/RoleImage/RoleFlippingImage/RoleFlippingImageSvgIcon/role-flipping-image-svg-icon.types";
import RoleFlippingImageSvgIcon from "~/components/shared/role/RoleImage/RoleFlippingImage/RoleFlippingImageSvgIcon/RoleFlippingImageSvgIcon.vue";

import { mountSuspendedComponent } from "~/tests/unit/utils/helpers/mount.helpers";

describe("Role Flipping Image Svg Icon Component", () => {
  let wrapper: ReturnType<typeof mount<typeof RoleFlippingImageSvgIcon>>;
  const defaultProps: RoleFlippingImageSvgIconProps = { svgIconPath: "/svg/werewolf.svg" };

  async function mountRoleFlippingImageSvgIconComponent(options: ComponentMountingOptions<typeof RoleFlippingImageSvgIcon> = {}):
  Promise<ReturnType<typeof mount<typeof RoleFlippingImageSvgIcon>>> {
    return mountSuspendedComponent(RoleFlippingImageSvgIcon, {
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountRoleFlippingImageSvgIconComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Size", () => {
    it("should set height to 75 when rendered.", () => {
      expect(wrapper.attributes("height")).toBe("75");
    });

    it("should set width to 75 when rendered.", () => {
      expect(wrapper.attributes("width")).toBe("75");
    });
  });
});