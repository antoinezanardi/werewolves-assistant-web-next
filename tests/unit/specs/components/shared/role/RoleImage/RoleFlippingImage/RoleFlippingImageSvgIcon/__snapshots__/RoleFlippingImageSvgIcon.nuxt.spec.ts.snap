// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Role Flipping Image Svg Icon Component > should match snapshot when rendered. 1`] = `"<nuxt-img-stub alt="Role icon" height="75" src="/svg/werewolf.svg" width="75" preload="false" crossorigin="false" placeholder="/svg/misc/infinite-spinner.svg" id="role-flipping-image-svg-icon" class="absolute animate__bounceIn bottom-4 right-2"></nuxt-img-stub>"`;
